sdk/tables/azure-data-tables/azure/data/tables/_base_client.py

```diff
@@ -298,7 +298,10 @@ def _batch_send(self, table_name: str, *reqs: HttpRequest, **kwargs) -> List[Map
             _validate_tablename_error(decoded, table_name)
             raise decoded
 
-        parts = list(response.parts())
+        try:
+            parts = list(response.parts())
+        except IndexError:
+            return []
```

Comment:
I don't think we need the early return statement here, seems like the rest of the logic in this method shouldn't cause any problems with an empty list.
```suggestion
            parts = []
```

Reply:
Agree. But when `parts` is an empty list, the rest logic will do nothing. We should return in this case.